lib/internal/process/promises.js

```diff
@@ -99,3 +99,12 @@ function setupPromises(scheduleMicrotasks) {
 
   return emitPendingUnhandledRejections;
 }
+
+const savedString = String;
+function safeString(value) {
+  try {
+    return savedString(value);
+  } catch (err) {
+    return '<toString() threw exception>';
```

Comment:
If even the error thrown can not be stringified safely then IMO it's reasonable to crash with what we've got. Not including the error message makes it more confusing than what it is now if you have no idea whose `toString()` is throwing...